.github/workflows/lint.yml

```
@@ -14,6 +14,14 @@ permissions:
   contents: write
 
 jobs:
+  docker-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - uses: hadolint/hadolint-action@54c9adbab1582c2ef04b2016b760714a4bfde3cf # v3.1.0
```

Can we use the tag instead of the hash?

```suggestion
      - uses: actions/checkout@4

      - uses: hadolint/hadolint-action@3.1.0
```

---

Saw https://github.com/mermaid-js/mermaid/pull/5764, great!